src/napari_matplotlib/histogram.py

```diff
@@ -48,7 +53,7 @@ def draw(self) -> None:
 
         # Important to calculate bins after slicing 3D data, to avoid reading
         # whole cube into memory.
-        bins = np.linspace(np.min(data), np.max(data), 100)
+        bins = np.linspace(np.min(data), np.max(data), 100, dtype=data.dtype)
```

Comment:
I'm not sure this is the right fix, since it results in un-even bin sizes if max - min of the data isn't a multiple of 100:
```python
>>> np.linspace(1, 150, 100, dtype=int)
array([  1,   2,   4,   5,   7,   8,  10,  11,  13,  14,  16,  17,  19,
        20,  22,  23,  25,  26,  28,  29,  31,  32,  34,  35,  37,  38,
        40,  41,  43,  44,  46,  47,  49,  50,  52,  53,  55,  56,  58,
        59,  61,  62,  64,  65,  67,  68,  70,  71,  73,  74,  76,  77,
        79,  80,  82,  83,  85,  86,  88,  89,  91,  92,  94,  95,  97,
        98, 100, 101, 103, 104, 106, 107, 109, 110, 112, 113, 115, 116,
       118, 119, 121, 122, 124, 125, 127, 128, 130, 131, 133, 134, 136,
       137, 139, 140, 142, 143, 145, 146, 148, 150])
```

I think the better thing to do here is something like:

```python
if data.dtype.kind == 'i':
    # Make sure integer data types have integer sized bins
    step = (np.max(data) -  np.min(data)) // 100
    bins = np.arange(np.min(data), np.max(data) + step, step)
```

Reply:
ooo well spotted! I didn't think to check that

Reply:
[`mypy` doesn't like this change](https://results.pre-commit.ci/run/github/486960190/1704984716.nV8PbjYAQRCE3pXbcJoxJQ) when passing the bins to `self.axes.hist`. Passing `bins.tolist()` to it instead seems to be okay, but is there a better way?

Reply:
👍 I think this is the best way